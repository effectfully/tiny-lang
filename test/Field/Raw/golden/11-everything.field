let ?a = T;
let ?b = F;
let #v = {T, F, T};
let ?or = ?a or ?b;
let ?and = ?or or ?b;
let ?xor = ?and or ?or;
let f = 0;
let g = 1;
let h = 2;
let ?feq = g == f;
let ?fle = h <= f;
let ?flt = f < g;
let ?fge = g >= h;
let ?fgt = h > f;
let add = f + g;
let sub = g - h;
let mul = h * f;
let div = f / g;
let ?bat = #v[f];
let ?not = not ?a;
let ?neq0 = neq0 f;
let neg' = neg g;
let inv' = inv h;
let #unpack = unpack h;
let ?let = T;
assert T;
for j = 0 to 2 do end;
let ?if = if T then T else T;
let asf = 1 : field;
let ?as = T : bool;
let #as = { T } : vector;