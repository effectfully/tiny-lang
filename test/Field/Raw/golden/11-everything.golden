Program [(Var "eb",Some Bool),(Var "ef",Some Field),(Var "ev",Some Vector)] Statements [ELet (Var "a",Some Bool) (EConst (Some (UniConst Bool True))),ELet (Var "b",Some Bool) (EConst (Some (UniConst Bool False))),ELet (Var "or'",Some Bool) (EAppBinOp Or (EVar (Var "a")) (EVar (Var "b"))),ELet (Var "or''",Some Bool) (EAppBinOp Or (EVar (Var "a")) (EVar (Var "eb"))),ELet (Var "and'",Some Bool) (EAppBinOp And (EVar (Var "or''")) (EVar (Var "b"))),ELet (Var "xor'",Some Bool) (EAppBinOp Xor (EVar (Var "and'")) (EVar (Var "or''"))),ELet (Var "f",Some Field) (EConst (Some 0)),ELet (Var "g",Some Field) (EConst (Some 1)),ELet (Var "h",Some Field) (EConst (Some 2)),ELet (Var "feq",Some Bool) (EAppBinOp FEq (EVar (Var "g")) (EVar (Var "f"))),ELet (Var "fe'",Some Bool) (EAppBinOp FEq (EVar (Var "ef")) (EVar (Var "f"))),ELet (Var "fle",Some Bool) (EAppBinOp FLe (EVar (Var "h")) (EVar (Var "f"))),ELet (Var "flt",Some Bool) (EAppBinOp FLt (EVar (Var "f")) (EVar (Var "g"))),ELet (Var "fge",Some Bool) (EAppBinOp FGe (EVar (Var "g")) (EVar (Var "h"))),ELet (Var "fgt",Some Bool) (EAppBinOp FGt (EVar (Var "h")) (EVar (Var "f"))),ELet (Var "add",Some Field) (EAppBinOp Add (EVar (Var "f")) (EVar (Var "g"))),ELet (Var "ad'",Some Field) (EAppBinOp Add (EVar (Var "f")) (EVar (Var "ef"))),ELet (Var "sub",Some Field) (EAppBinOp Sub (EVar (Var "g")) (EVar (Var "h"))),ELet (Var "mul",Some Field) (EAppBinOp Mul (EVar (Var "h")) (EVar (Var "f"))),ELet (Var "div",Some Field) (EAppBinOp Div (EVar (Var "f")) (EVar (Var "g"))),ELet (Var "v",Some Vector) (EConst (Some (UniConst Vector [True,False,True]))),ELet (Var "bat",Some Bool) (EAppBinOp BAt (EVar (Var "f")) (EVar (Var "v"))),ELet (Var "ba'",Some Bool) (EAppBinOp BAt (EVar (Var "f")) (EVar (Var "ev"))),ELet (Var "not'",Some Bool) (EAppUnOp Not (EVar (Var "a"))),ELet (Var "neq0'",Some Bool) (EAppUnOp Neq0 (EVar (Var "f"))),ELet (Var "neg'",Some Field) (EAppUnOp Neg (EVar (Var "g"))),ELet (Var "inv'",Some Field) (EAppUnOp Inv (EVar (Var "h"))),ELet (Var "unp",Some Vector) (EAppUnOp Unp (EVar (Var "h"))),ELet (Var "let'",Some Bool) (EConst (Some (UniConst Bool True))),EAssert (EConst (Some (UniConst Bool True))),EFor (Var "j") 0 2 (Statements []),ELet (Var "if'",Some Bool) (EIf (EConst (Some (UniConst Bool True))) (EConst (Some (UniConst Bool True))) (EConst (Some (UniConst Bool True)))),ELet (Var "asf",Some Field) (ETypeAnn (Some Field) (EConst (Some 1))),ELet (Var "asb",Some Bool) (ETypeAnn (Some Bool) (EConst (Some (UniConst Bool True)))),ELet (Var "asv",Some Vector) (ETypeAnn (Some Vector) (EConst (Some (UniConst Vector [True]))))]